### PR TITLE
Make `Executable` async using `tokio::process::Command`

### PR DESCRIPTION
**Pull request by future-highway**
`Executable` now uses `tokio::process::Command` for its async abilities. It no longer uses the `Process` enum that we have in the module as it seems like it's more trouble then it's worth.

`NestedAuraed` is still synchronous as I would rather see how that evolves to fill our needs prior to making it async. It also still uses `Process`, but there is probably no reason for `Process` to exist separately so it will probably be incorporated into `NestedAuraed`.

I also refactored LogChannel into a more expected structure.

### CODE REVIEW

File: auraed/src/logging/log_channel.rs [hunk 1]
```diff
@@ -31,65 +31,41 @@
...
+        self.tx.subscribe()
     }
 
     /// Wrapper that sends a log line to the channel
-    pub fn log_line(producer: Sender<LogItem>, line: String) {
+    pub fn send(&self, line: String) {
         // send returns an Err if there are no receivers. We ignore that.
-        let _ = producer.send(LogItem {
+        let _ = self.tx.send(LogItem {
             channel: "unknown".to_string(),
```
Comment (future-highway) [inline auraed/src/logging/log_channel.rs:63]: Should this be the LogChannel's name?

**Review by krisnova — DISMISSED** [review]

File: auraed/src/logging/log_channel.rs (hunk 1 above)
Comment (dmah42) [reply auraed/src/logging/log_channel.rs:63]: yes!

File: auraed/src/runtime/cell_service/executables/executable.rs [hunk 2]
```diff
@@ -47,168 +39,89 @@ impl Executable {
...
-        let process = exec(command)?;
+        let mut child = command
+            .current_dir("/")
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()?;
+
+        let stdout = child.stdout.take().expect("stdout");
+        let log_channel = LogChannel::new(format!("{}::stdout", self.name));
+        let span = info_span!("running process", name = ?self.name);
```
Comment (dmah42) [inline auraed/src/runtime/cell_service/executables/executable.rs:60]: I think this should be inside the tokio spawn thread?

**Review by dmah42 — APPROVED** [review]
I'm curious if this resolves the delay on stop.

File: auraed/src/runtime/cell_service/executables/executable.rs (hunk 2 above)
Comment (dmah42) [reply auraed/src/runtime/cell_service/executables/executable.rs:60]: ah gotcha. you're right, of course.  i keep forgetting the extra "entered" call that's necessary to enable the span.

**Review by dmah42 — APPROVED** [review]
<3 this change

**Comment by future-highway** [comment]
> I'm curious if this resolves the delay on stop.

I think yes and no. Assuming the sleep call was the cause, then we would no longer have an "unnecessary" delay. However, this should read stdout/err to the end, so it will take whatever time that takes for `kill` to return.

**Pull request merged**